conformance/failure_list_trie_node.cc

```diff
@@ -37,28 +37,38 @@ absl::Status FailureListTrieNode::Insert(absl::string_view test_name) {
 
 void FailureListTrieNode::InsertImpl(absl::string_view test_name) {
   absl::string_view section = test_name.substr(0, test_name.find('.'));
+  
+  bool name_equals_section = test_name == section;
 
-  // Extracted last section -> no more '.' -> test_name_copy will be equal to
-  // section
-  if (test_name == section) {
-    children_.push_back(std::make_unique<FailureListTrieNode>(section));
-    return;
-  }
-  test_name = test_name.substr(section.length() + 1);
+  // test_name cannot be overwritten
+  absl::string_view test_name_rest =
+      name_equals_section ? "" : test_name.substr(section.length() + 1) ;
   for (auto& child : children_) {
-    if (child->data_ == section) {
-      return child->InsertImpl(test_name);
+    // Extracted last section -> no more '.' -> test_name will be equal to
+    // section
+    if (child->data_ == section && name_equals_section) {
```

Comment:
Maybe these two if statements could be simplified a little bit by combining them like this:
```
if (child->data_ == section) {
  if (name_equals_section) {
    child->is_test_name = true;
  } else {
    child->InsertImpl(test_name_rest);
  }
  return;
}
```